test/parallel/test-url-pathtofileurl.js

```diff
@@ -22,3 +22,49 @@ const url = require('url');
   const fileURL = url.pathToFileURL('test/%').href;
   assert.ok(fileURL.includes('%25'));
 }
+
+{
+  const testCases = [
+    // lowercase ascii alpha
+    { path: '/foobar.mjs', expected: 'file:///foobar.mjs' },
+    // uppercase ascii alpha
+    { path: '/FOOBAR.mjs', expected: 'file:///FOOBAR.mjs' },
+    // space
+    { path: '/foo bar.mjs', expected: 'file:///foo%20bar.mjs' },
+    // question mark
+    { path: '/foo?bar.mjs', expected: 'file:///foo%3Fbar.mjs' },
+    // number sign
+    { path: '/foo#bar.mjs', expected: 'file:///foo%23bar.mjs' },
+    // ampersand
+    { path: '/foo&bar.mjs', expected: 'file:///foo&bar.mjs' },
+    // equals
+    { path: '/foo=bar.mjs', expected: 'file:///foo=bar.mjs' },
+    // colon
+    { path: '/foo:bar.mjs', expected: 'file:///foo:bar.mjs' },
+    // semicolon
+    { path: '/foo;bar.mjs', expected: 'file:///foo;bar.mjs' },
+    // percent
+    { path: '/foo%bar.mjs', expected: 'file:///foo%25bar.mjs' },
+    // backslash
+    { path: '/foo\\bar.mjs', expected: isWindows ? 'file:///foo/bar.mjs' : 'file:///foo%5Cbar.mjs' },
```

Comment:
This line is causing failure on Windows. I wrote the Windows result using `path.win32`, but I'm on Linux so it did not add a drive letter. Windows CI produces `file:///c:/foobar.mjs`. I don't really like to hardcode the drive letter because it's dependent on the cwd: if someone tries to build Node on Windows on a different drive, it may fail.

My idea would be to skip this test on Windows (it's only relevant on non-windows systems), but maybe you know how to get the correct drive letter reliably.